test.js

```diff
@@ -138,3 +138,26 @@ test('ipc stream api wrapped', async (t) => {
     }
   }
 })
+
+test('ipc client close when heartbeat fails', async (t) => {
+  t.plan(1)
+  // t.timeout(8000)
+  const server = new IPC({
+    socketPath,
+    handlers: { start: (params) => params.result }
+  })
+  t.teardown(() => server.close())
+  const client = new IPC({
+    socketPath: 'test.sock',
+    connect: true
+  })
+
+  client.once('close', () => {
+    t.pass('client closed by server when heartbeat fails')
+  })
+
+  await server.ready()
+  await client.ready()
+
+  client._beat = () => {} // simulate heartbeat failure
```

Comment:
More of a question: doesn't the test also work if we destroy the client after the first successful heartbeat? That way the test is closer to the actual flow we try to detect

Reply:
@HDegroote that assumes the zombies are created when clients are destroyed, whereas this assumes nothing about why the hearbeat has gone silent - renderer crashes etc might not include a call to destroy

Reply:
Makes sense!
Optional: For a slight modification of the test, you could first check that the heartbeat was received at least once, that way the happy path is also explicitly tested

Reply:
@HDegroote good suggestion, test modified and it did solve a false-positive :+1: